<?xml version="1.0"?>

<launch>

    <arg name="world_name" default="basic"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find ORTINa)/worlds/demo_world.world"/>
        <arg name="paused" value="true"/>
        <arg name="verbose" value="true"/>
    </include>

    <group ns="uav1">
        <include file="$(find ORTINa)/launch/dji_m100_spawn.launch">
        <arg name="name" value="uav1" />
        <arg name="tf_prefix" value="uav1" />
        <arg name="is_rotors_controlled" value="true"/>
        <arg name="is_hitl_model" value="false"/>
        </include>
    </group>

    <!--Spawn LQ-Servo Controller-->
    <node pkg="ORTINa" type="lq_servo_node" name="lq_servo_node" output="screen"/>

    <!-- Action Server-->
    <node pkg="ORTINa" type="pose_action_server" name="pose_action_server"/>
    <node pkg="ORTINa" type="pose_action_client" name="pose_action_client"/>

    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ORTINa)/rviz_config/config.rviz"/> -->

</launch>  